### ISSUE TITLE
Adding a NO GUTTER class

### ISSUE THREAD

**Issue by rokkitpress** [twbs/bootstrap#19834]
Hi,

During a recent mock-up of a site using V4 alpha 2 it was required to remove the gutter on particular columns allowing the content inside the column to issue the padding and the two columns to line up side by side with no space between.

I created the classes `.ng`, `.ng-[$break-point]-l`, `ng-[$break-point]-r`, these have actually come in pretty useful and with the use of Sass to; `$enable-no-gutters : true/false`, it could also come in handy depending on the build requirements.

To make it truly useful I feel it would need to have the same principles as the "Available classes" on the "Responsive utilities" page of the docs. Again if this is opt in or out it is down to the designer or developer if they are happy with the extra code.

My reason for removing the the padding is that on smaller screens to make certain components optimized I did not want the double padding from the column and then the component.

I had also used the classes again when adding some block buttons to the bottom of a column that I wanted stretched to either side of the column.

[You can take a look at the project here.](http://codepen.io/Craig-Watson/pen/vGVbVK/)

It is just a rough mock-up so excuse and unused classes or unfinished business as I will still be working on it in the coming days.

Apologies if this is of topic im still getting used to github.

Craig

**Comment by twbs-lmvtfy** [comment]
Hi @rokkitpress!

You appear to have posted a live example (http://codepen.io/Craig-Watson/pen/vGVbVK.html), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 8, column 1 thru column 20: Start tag ```` body ```` seen but an element of the same type was already open.
* line 8, column 1 thru column 20: Cannot recover after last error. Any further errors will be ignored.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by rokkitpress** [comment]
Removed body tags from codepen

[New Link](http://codepen.io/Craig-Watson/pen/vGVbVK/)

**Comment by twbs-lmvtfy** [comment]
Hi @rokkitpress!

You appear to have posted a live example (http://codepen.io/Craig-Watson/pen/vGVbVK.html), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 52, column 2 thru column 91: Bad value ```` button ```` for attribute ```` type ```` on element ```` a ````: Subtype missing.
* line 53, column 2 thru column 91: Bad value ```` button ```` for attribute ```` type ```` on element ```` a ````: Subtype missing.
* line 54, column 2 thru column 91: Bad value ```` button ```` for attribute ```` type ```` on element ```` a ````: Subtype missing.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by rokkitpress** [comment]
Well apparently my statement about this being a "rough" mock up was the understatement of the year so far... 

[NEW LINK>>>](http://codepen.io/Craig-Watson/pen/vGVbVK/?editors=1100)

**Comment by twbs-lmvtfy** [comment]
Hi @rokkitpress!

You appear to have posted a live example (http://codepen.io/Craig-Watson/pen/vGVbVK.html), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 233, column 7: [E041](https://github.com/twbs/bootlint/wiki/E041): `.carousel-inner` must have exactly one `.item.active` child.
* line 10, column 1: [W012](https://github.com/twbs/bootlint/wiki/W012): `.navbar`'s first child element should always be either `.container` or `.container-fluid`

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by RyanZim** [comment]
@rokkitpress Sounds like a good suggestion to me; a few things should be noted:

- The classes should probably be renamed as it is rather ambiguous whether `.ng-md-l` removes the gutter at `md` only, `md` and up, or `md` and down. Perhaps something in keeping with `.hidden-*-up` and `.hidden-*-down`?
- If we go with such a naming scheme, the `.ng` will not be needed. (Redundant with `.ng-xs-*-up`)

Of course, the core team needs to decide if they think this is a good feature or not before we go too much further.

**Comment by rokkitpress** [comment]
@RyanZim thanks for the support on this I was not sure how it would go down. I did mention the the page [Responsive utilities](http://v4-alpha.getbootstrap.com/layout/responsive-utilities/) in my initial comment to find out what the thoughts were on using such classes like `.ng-xs-r-up`. 

Hopefully they will be interested as it would come in quite handy I'm sure.

**Comment by RyanZim** [comment]
`.ng-[a, l, r]-[breakpoint]-[up, down]` would make more sense to me. (Breakpoint and `up` or `down` beside each other.)
Here is a JSFiddle demonstrating `.ng-*-xs-up`: https://jsfiddle.net/kqeev2x6/1/.

**Comment by takuhii** [comment]
Actually, this would be really handle, I was using the new .CARD classes in
BootStrap V4 and wanted to change the gutter between the cards, so they
were a little closer together for what I wanted to do. I found that I
couldn't really do this, so something like this would be great...

**Comment by cvrebert** [comment]
Approximate duplicate of #19107.

**Comment by RyanZim** [comment]
Hey, just thought about it, such a class already exists: `.p-x-0`.

**Comment by rokkitpress** [comment]
@RyanZim actually yes I have been enjoying the use of those classes to... maybe what we need is the option on extend them with the breakpoint feature?

It seems silly now to entertain the thought of a whole new set of classes when the potential is already there to make existing ones do the job..

What do you think?

**Comment by RyanZim** [comment]
@rokkitpress It's probably best to open a new issue for that.

I also left a comment at https://github.com/twbs/bootstrap/issues/19107#issuecomment-216677324.
